src/Microsoft.OpenApi/Services/OpenApiWalker.cs

```diff
@@ -46,12 +48,15 @@ public void Walk(OpenApiDocument doc)
         /// <param name="tags"></param>
         internal void Walk(IList<OpenApiTag> tags)
         {
+            _visitor.Enter(OpenApiConstants.Tags);
             _visitor.Visit(tags);
 
             foreach (var tag in tags)
             {
                 Walk(tag);
             }
```

Comment:
We should add the array index to get the correct json pointer here.

                for ( int i =0; i<tags.Count(); i++ )
                {
                    _visitor.Enter(i);
                    Walk(tags[i]);
                     _visitor.Exit();
                }

Also, this raises another point that the Enter call should be the responsibility of the caller. In this case, the Walk(tags[i]) should NOT need to call any enter for the Tag object itself (since the array index is used instead).